### PR TITLE
Allow :pid to be passed in as metadata in string format

### PR DESCRIPTION
**Pull request by anthonyfalzetti**
Allow :pid to be passed in as metadata in string format

### CODE REVIEW

**Comment by anthonyfalzetti** [comment]
@adherr they were actual process pids. Which was causing the encoder to fail. This simply returns the string version of the pid which will make tracking a single process down possible. 
```elixir
json_encoder.encode!(%{pid: #PID<0.1462.0>})
** (Poison.EncodeError) unable to encode value: #PID<0.1462.0>
```

**Comment by fieldinrain** [comment]
hi, @anthonyfalzetti 
Thank you for your pull request.
Now we are using Jason as the default json encoder from 0.1.9,  and we'd be glad to see your `allow pid` pull request.

**Comment by anthonyfalzetti** [comment]
@fieldinrain I added a test to show the functionality.

**Comment by fieldinrain** [comment]
hi, @anthonyfalzetti
The following two tests failed, could you check them?  
```
 1) test doesn't trim metadata (LoggerJSONFileBackendTest)
     test/logger_json_file_backend_test.exs:47
     ** (MatchError) no match of right hand side value: {:error, :bad_module}
     code: json_log = Poison.decode! log()
     stacktrace:
       test/logger_json_file_backend_test.exs:76: LoggerJSONFileBackendTest.path/0
       test/logger_json_file_backend_test.exs:81: LoggerJSONFileBackendTest.log/0
       test/logger_json_file_backend_test.exs:50: (test)



  2) test can log structured object (LoggerJSONFileBackendTest)
     test/logger_json_file_backend_test.exs:29
     ** (MatchError) no match of right hand side value: {:error, :bad_module}
     code: json_log = Jason.decode! log()
     stacktrace:
       test/logger_json_file_backend_test.exs:76: LoggerJSONFileBackendTest.path/0
       test/logger_json_file_backend_test.exs:81: LoggerJSONFileBackendTest.log/0
       test/logger_json_file_backend_test.exs:31: (test)
```

**Comment by anthonyfalzetti** [comment]
Oh wow, I broke a ton of tests. I will dig in tonight

File: lib/logger_json_file_backend.ex [hunk 1]
```diff
@@ -63,6 +63,7 @@ defmodule LoggerJSONFileBackend do
         true ->
           json_encoder.encode!(Map.merge(%{level: level, uuid: UUID.uuid4(), message: (msg |> IO.iodata_to_binary), time: format_time(ts)}, take_metadata(md, keys, triming))) <> "\n"
         false ->
+          metadata = take_metadata(md, keys, triming)
```
Comment (fieldinrain) [inline lib/logger_json_file_backend.ex]:
```
variable "metadata" is unused
```
Could you check it?

Comment (anthonyfalzetti) [reply lib/logger_json_file_backend.ex]: removed, and verified that `mix test test` still passes

**Review by fieldinrain — APPROVED** [review]

**Pull request merged**